test/test_nn.py

```diff
@@ -2606,6 +2607,61 @@ def test_orthogonal(self):
                                          torch.eye(rows) * gain ** 2, prec=1e-6)
 
 
+# functional loss wrappers
+
+
+class _SmoothL1LossFunctional(_Loss):
```

Comment:
I added this interface to test these functional functions because they are not called in their corresponding `Module`. 

The Module uses the class name to get the function directly from the backend (as https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/loss.py#L22 and https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/loss.py#L33)

I don't know what is the best way to do it: (1) using this "fake modules" to test, (2) update the nn.loss to actually call the functional instead of using the class name to get the function from the backend (as currently implemented) or (3) something else.